model/src/cfd.rs

```diff
@@ -3131,180 +3134,195 @@ mod tests {
 
     #[test]
     fn given_current_settlement_in_12_hours_and_candidate_in_19_then_7_hour_extension() {
-        let current_event_id =
-            BitMexPriceEventId::with_20_digits(OffsetDateTime::now_utc() + 12.hours());
-        let candidate_event_id =
-            BitMexPriceEventId::with_20_digits(OffsetDateTime::now_utc() + 19.hours());
+        for now in common_time_boundaries() {
```

Comment:
I'm personally not a fan of having loops in tests. 
I do prefer dedicated tests per values because if one of the asserts fails, you wouldn't know immediately which one it is. 

If you want to avoid code duplication you can extract parts into an own function and just pass in now.

Reply:
I totally agree - I went for this as a "quick patch". I don't assume these tests will change much, but I wanted to make sure we test the boundaries. 

> If you want to avoid code duplication you can extract parts into an own function and just pass in now. 

I can do that, but the assert would still have to be by test to have value. 
I think the proper way forward might configuring proptests for these.

I can add more tests, but I am not happy with this ever growing file 😅
I might just write a ticket and come back to this after some more thought - will decide tomorrow :)

Reply:
As solution for now I added `now` to be printed in the assert so we know with what value a test failed if it does.

Additionally I added https://github.com/itchysats/itchysats/issues/2116 as follow up